### PR TITLE
Add net.jockx.Kulki

### PR DESCRIPTION
**Pull request by pietrygamat**
Kulki is a simple game where the player moves marbles around the board in order to match rows of the same color. 

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: net.jockx.Kulki.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>net.jockx.Kulki</id>
+
+    <name>Kulki</name>
+    <summary>Simple logic game of matching marbles</summary>
+    <icon type="stock">net.jockx.Kulki</icon>
+    <icon type="remote">https://kulki.jockx.net/img/favicon.png</icon>
```
Comment (hfiguiere) [inline net.jockx.Kulki.metainfo.xml:8]:
```suggestion
```

neither icon type work.

File: net.jockx.Kulki.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    buildsystem: simple
+    build-commands:
+      - cp -r bin/. $FLATPAK_DEST/bin
+      - cp -r lib/. $FLATPAK_DEST/lib
+      - install -Dm644 "${FLATPAK_ID}.desktop" -t "/app/share/applications"
+      - install -Dm644 "${FLATPAK_ID}.png" -t "/app/share/icons/hicolor/512x512/apps/"
+      - install -Dm644 "${FLATPAK_ID}.metainfo.xml" -t "${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml"
+    sources:
+      - type: archive
+        url: https://github.com/jockx-net/kulki/releases/download/v1.0.0/Kulki-1.0.0.tar.gz
```
Comment (hfiguiere) [inline net.jockx.Kulki.yml:20]: how will this work on multiple architectures? I know that JAVA is compile once debug everywhere, but the JDK surely ain't.

File: net.jockx.Kulki.png
Comment (hfiguiere) [inline net.jockx.Kulki.png:1]: the icon should be managed in the upstrem repository. and be part of the distribution tarball.

File: net.jockx.Kulki.metainfo.xml
Comment (hfiguiere) [inline net.jockx.Kulki.metainfo.xml:1]: the appstream file should be managed in the upstrem repository. and be part of the distribution tarball.

File: net.jockx.Kulki.desktop
Comment (hfiguiere) [inline net.jockx.Kulki.desktop:1]: the desktop file should be managed in the upstrem repository. and be part of the distribution tarball.

File: net.jockx.Kulki.desktop [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+Type=Application
+
+Name=Kulki
+Comment=Simple logic game of matching marbles
+Categories=Game;LogicGame;
+
+Icon=net.jockx.Kulki
+Exec=Kulki
+Terminal=false
+X-Desktop-File-Install-Version=0.26
```
Comment (hfiguiere) [inline net.jockx.Kulki.desktop:12]:
```suggestion
```

File: net.jockx.Kulki.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>MIT</project_license>
+
+    <developer id="pietrygamat">
```
Comment (bbhtt) [inline net.jockx.Kulki.metainfo.xml:13]: Should be reverse DNS.

File: net.jockx.Kulki.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        <screenshot type="default">
+            <image>https://kulki.jockx.net/img/screenshot.png</image>
+        </screenshot>
+    </screenshots>
+
+    <releases>
+        <release version="1.0.0" date="2024-05-11"></release>
+    </releases>
+
+    <content_rating type="oars-1.0" />
```
Comment (bbhtt) [inline net.jockx.Kulki.metainfo.xml:43]:
```suggestion
    <content_rating type="oars-1.1" />
```

File: net.jockx.Kulki.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+command: Kulki
+finish-args:
+  - --socket=x11
+  - --share=ipc
+modules:
+  - name: kulki
+    buildsystem: simple
+    build-commands:
+      - cp -r bin/. $FLATPAK_DEST/bin
+      - cp -r lib/. $FLATPAK_DEST/lib
```
Comment (bbhtt) [inline net.jockx.Kulki.yml:14]: You have to mkdir these $FLATPAK_DEST/{bin, lib, share,...} etc.

File: net.jockx.Kulki.yml (hunk 2 above)
Comment (pietrygamat) [reply net.jockx.Kulki.yml:20]: This app is not multiplaform. It's a amd64 release only at this point. Should I specifiy that somewhere explicitly?

Comment (hfiguiere) [reply net.jockx.Kulki.yml:21]: so it builds a broken aarch64 flatpak.

Comment (bbhtt) [reply net.jockx.Kulki.yml:21]: You need a flathub.json file https://docs.flathub.org/docs/for-app-authors/maintenance#limiting-the-set-of-architectures-to-build-on

**Comment by bbhtt** [comment]
bot, build net.jockx.Kulki

File: net.jockx.Kulki.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+id: net.jockx.Kulki
```
Comment (bbhtt) [inline net.jockx.Kulki.yml:1]: Can't access https://jockx.net/

image

Comment (bbhtt) [reply net.jockx.Kulki.yml:1]: Ah it is https://kulki.jockx.net/

**Pull request closed without merging**